### PR TITLE
Implement MapGenerator::updateTreeSeasons

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/OpenLoco/src/Map/MapGenerator.cpp [hunk 1]
```diff
@@ -878,6 +878,32 @@ namespace OpenLoco::World::MapGenerator
...
+            auto tile = TileManager::get(pos);
+            for (auto el : tile)
+            {
+                auto* treeEl = el.as<TreeElement>();
+                if (treeEl == nullptr)
+                {
+                    continue;
+                }
+
+                if (treeEl->season() < 4 && treeEl->unk6_80())
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Map/MapGenerator.cpp:897]: !treeEl->unk6_80()

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Map/MapGenerator.cpp:897]:
```suggestion
                if (treeEl->season() < 4 && !treeEl->unk6_80())
```

**Review by LeftofZen — APPROVED** [review]
Functionally tested by loading a swiss alpa scenario and fastforwarding the game through a full year in both this branch and master - tree behaviour seems identical to master and trees get snowy and unsnowy at the correct times.

**Comment by duncanspumpkin** [comment]
> You've got the right idea, but this function is only applied in the map generator. It is applied right after generating a new landscape.

Some scenarios use the map generator. Not sure if Swiss does though.

**Review by ZehMatt — APPROVED** [review]

**Comment by LeftofZen** [comment]
> You've got the right idea, but this function is only applied in the map generator. It is applied right after generating a new landscape.

Apologies for the oversight.

Testing in the map gen, seems to work:
image

**Pull request merged**